### PR TITLE
Homework 2

### PR DESCRIPTION
**Pull request by DimitriIlnitski**
Homework-2 completed

### CODE REVIEW

File: src/common/Input/Input.jsx [hunk 1]
```diff
@@ -4,10 +4,8 @@ function Input(props) {
 	const {
 		labelText,
 		className,
-		// labelClass,
 		placeholderText,
 		inputName,
-		// inputClassName,
```
Comment (KitTheProgrammer) [inline src/common/Input/Input.jsx:10]: don't forget to remove those kinds of comments

File: src/components/CourseInfo/CourseInfo.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import './CourseInfo.css';
+
+import React, { useContext } from 'react';
+import { Link, useParams } from 'react-router-dom';
+import { AppContext, formatTime, getAuthorsList } from '../../helpers';
+
+function CourseInfo() {
+	const { courseList, authorsList } = useContext(AppContext);
+	const { id } = useParams();
+	const course = courseList.find((item) => item.id === id);
```
Comment (KitTheProgrammer) [inline src/components/CourseInfo/CourseInfo.jsx]: use `useMemo` here

File: src/components/Registration/Registration.jsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+	const handleChange =
+		(setter) =>
+		({ target: { value } }) => {
+			setter(value);
+		};
+
+	const handleChangeName = handleChange(setName);
+	const handleChangeEmail = handleChange(setEmail);
+	const handleChangePassword = handleChange(setPassword);
```
Comment (KitTheProgrammer) [inline src/components/Registration/Registration.jsx:26]: a bit overkill here. You can just use controlled input (you can find information about it in the React docs)

**Review by KitTheProgrammer — CHANGES_REQUESTED** [review]
While testing your app I found:
1. When first entering the app I was not redirected to login, I can access any app page without being logged it.
2. If you write the address of the course yourself (http://localhost:3000/courses/de5aaa59-90f5-4dbc-b8a9-aaf205c551ba) the app crashes. The same is happening with random ID
3. I can create authors with name containing only spaces.
4. I didn't find exact causes of this, but "logout" button disappears sometimes

File: src/components/CourseInfo/CourseInfo.jsx (hunk 2 above)
Comment (DimitriIlnitski) [reply src/components/CourseInfo/CourseInfo.jsx]: Done

File: src/common/Input/Input.jsx (hunk 1 above)
Comment (DimitriIlnitski) [reply src/common/Input/Input.jsx:10]: Have deleted all comments.
Have created protected routes, so you can not access pages directly.
Now you can not create author with just spaces - placed patter, which expect min 2 letters with extry letter to be in upper case.
About Logout button - our task forsee, that on login and registration pages you an not see user name and logout button. On all other pages you can see it.

File: src/components/Registration/Registration.jsx (hunk 3 above)
Comment (DimitriIlnitski) [reply src/components/Registration/Registration.jsx:26]: Previous time then I have had 3 states, you advise me to create one function, which will handle states setting. Im not sure what shuld I do now. I see the following variant: I can either return 3 setting functions or can put general handleChange to helpers forlder and use it in all cases I have similar situation. Can you please guide me what to do.

**Review by KitTheProgrammer — APPROVED** [review]

**Comment by KitTheProgrammer** [comment]
25

**Pull request merged**